src/System.Reflection.Metadata/src/System/Reflection/Metadata/Handles.cs

```diff
@@ -7,77 +7,179 @@
 
 namespace System.Reflection.Metadata
 {
+    /// <summary>
+    /// Represents any metadata entity (type reference/definition/specification, method definition, custom attribute, etc.) or value (string, blob, guid, user string).
+    /// </summary>
+    /// <remarks>
+    /// Use <see cref="Handle"/> to store multiple kinds of handles.
+    /// </remarks>
     public struct Handle : IEquatable<Handle>
     {
-        internal readonly uint value;
+        private readonly int _value;
 
-        internal Handle(uint value)
+        // bits:
+        //    7: IsVirtual
+        // 0..6: token type
+        private readonly byte _vType;
```

Comment:
From an alternative perspective:

Why do we need to keep `Handle` around? Can we get away with only `EntityHandle`, and just always treat the other kinds of tokens separately in code? If so, can we remove `Handle` and rename `EntityHandle` to `Handle`. In other words, can we solve this problem without introducing new types by simply removing the ability to convert string, blob, and guid handles to `Handle`?